components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/cdc/CDCEventOutput.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.inbound.endpoint.protocol.cdc;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.debezium.engine.ChangeEvent;
+
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.AFTER;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.BEFORE;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.DB;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.OP;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.PAYLOAD;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.SOURCE;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.TABLE;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.TS_MS;
+
+public class CDCEventOutput {
+
+    private JsonObject payload;
+
+    private enum operations {c, r, u, d};
+
+    CDCEventOutput(ChangeEvent event) {
+        String valueString = event.value().toString();
+        JsonObject value = new Gson().fromJson(valueString, JsonObject.class);
+        this.payload = value.getAsJsonObject(PAYLOAD);
+    }
+
+    public JsonElement getJsonPayloadBeforeEvent() {
+        JsonElement beforeObject = null;
+        if (payload.has(BEFORE)) {
+            beforeObject = payload.get(BEFORE);
+        }
+        return beforeObject;
+
```

Comment:
Since [JsonObject](https://www.javadoc.io/doc/com.google.code.gson/gson/2.9.0/com.google.gson/com/google/gson/JsonObject.html)s `get` method will return null if member doesn't exists we don't need these check here.
```suggestion
        return payload.get(BEFORE);
```

Reply:
fixed